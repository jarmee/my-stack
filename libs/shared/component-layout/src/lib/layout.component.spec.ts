import { ComponentFixture, TestBed } from "@angular/core/testing";
import { By } from "@angular/platform-browser";
import { NoopAnimationsModule } from "@angular/platform-browser/animations";

import { LayoutComponent } from "./layout.component";

const CssSelectors = {
  ToggleNavigationButton: By.css('[data-test-id="navigtationButton"]'),
  Sidebar: By.css('[data-test-id="sidebar"]'),
};

describe("LayoutComponent", () => {
  let component: LayoutComponent;
  let fixture: ComponentFixture<LayoutComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [LayoutComponent, NoopAnimationsModule],
    }).compileComponents();

    fixture = TestBed.createComponent(LayoutComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it("should create", () => {
    expect(component).toBeTruthy();
  });

  it("should match snapshot", () => {
    expect(fixture).toMatchSnapshot();
  });

  describe("onToggleNavigation", () => {
    it("should open the navigation", () => {
      component.toggleSidebarNavigation = false;

      fixture.debugElement
        .query(CssSelectors.ToggleNavigationButton)
        .triggerEventHandler("click", {});
      fixture.detectChanges();

      expect(
        fixture.debugElement.query(CssSelectors.Sidebar).componentInstance
          .opened
      ).toBe(true);
    });

    it("should close the navigation", () => {
      component.toggleSidebarNavigation = true;

      fixture.debugElement
        .query(CssSelectors.ToggleNavigationButton)
        .triggerEventHandler("click", {});
      fixture.detectChanges();

      expect(
        fixture.debugElement.query(CssSelectors.Sidebar).componentInstance
          .opened
      ).toBe(false);
    });
  });
});
