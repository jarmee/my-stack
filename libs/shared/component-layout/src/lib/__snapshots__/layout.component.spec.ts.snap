// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutComponent should match snapshot 1`] = `
<mys-layout
  __ngContext__={[Function Number]}
  title=""
  toggleSidebarNavigation="false"
>
  <mat-toolbar
    class="mat-toolbar mat-primary mat-toolbar-single-row"
    color="primary"
  >
    <div
      class="container mx-auto flex items-center"
    >
      <button
        aria-label="Open Navigation"
        class="mdc-icon-button mat-mdc-icon-button _mat-animation-noopable mat-unthemed mat-mdc-button-base"
        data-test-id="navigtationButton"
        mat-icon-button=""
      >
        <span
          class="mat-mdc-button-persistent-ripple mdc-icon-button__ripple"
        />
        <mat-icon
          aria-hidden="true"
          class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color"
          data-mat-icon-type="font"
          role="img"
        >
          menu
        </mat-icon>
        <span
          class="mat-mdc-focus-indicator"
        />
        <span
          class="mat-ripple mat-mdc-button-ripple"
          matripple=""
        />
        <span
          class="mat-mdc-button-touch-target"
        />
      </button>
      <span
        class="mr-4"
      >
      </span>
    </div>
  </mat-toolbar><main>
    <mat-drawer-container
      class="mat-drawer-container"
    >
      <div
        class="mat-drawer-backdrop"
      />
      <div
        aria-hidden="true"
        class="cdk-visually-hidden cdk-focus-trap-anchor"
      />
      <mat-drawer
        autofocus="false"
        class="mat-drawer ng-trigger ng-trigger-transform mat-drawer-over"
        data-test-id="sidebar"
        mode="over"
        style="box-shadow:none;visibility:hidden;0:box-shadow;1:visibility;"
        tabindex="-1"
      >
        <div
          cdkscrollable=""
          class="mat-drawer-inner-container"
        />
      </mat-drawer>
      <div
        aria-hidden="true"
        class="cdk-visually-hidden cdk-focus-trap-anchor"
      />
      <mat-drawer-content
        class="mat-drawer-content container mx-auto pt-3"
      />
    </mat-drawer-container>
  </main>
</mys-layout>
`;
